container/go/cmd/puller/puller.go

```diff
@@ -60,6 +62,9 @@ func pull(imgName, dstPath string, platform v1.Platform) {
 	if err != nil {
 		log.Fatalf("reading image %q: %v", ref, err)
 	}
+	if cachePath != "" {
+		i = cache.Image(i, cache.NewFilesystemCache(cachePath))
```

Comment:
You are not using `i` anywhere. Change this to just `cache.Image(i, cache.NewFilesystemCache(cachePath))`

Reply:
Sorry typo, supposed to be reassigning img, not i, will make sense once my tarball PR is merged.